ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -140,41 +157,28 @@ void handleStatement(
    *
    * @param commandId The ID of the statement that has been written to the command topic.
    */
-  public Future<CommandStatus> registerQueuedStatement(final CommandId commandId) {
-    statusStore.put(
-        commandId,
-        new CommandStatus(CommandStatus.Status.QUEUED, "Statement written to command topic")
-    );
-
-    CommandStatusFuture result;
+  public CommandStatusFuture registerQueuedStatement(final CommandId commandId) {
     synchronized (statusFutures) {
-      result = statusFutures.get(commandId);
-      if (result != null) {
-        return result;
+      if (statusFutures.containsKey(commandId)) {
```

Comment:
consider switching `statusFutures` to a `ConcurrentMap` too.  Also, use the atomic `compute` methods...

e.g.

```
statusFutures.compute(commandId, (key, oldValue) -> {
   if (oldValue != null) {
       throw blah
   }

   return new CommandStatusFuture(commandId, initialStatus);
});
```